File: Array/2176. Count Equal and Divisible Pairs in an Array.py
class Solution:
    def countPairs(self, nums: List[int], k: int) -> int:
        ans, n = 0, len(nums)
        for i in range(n):
            for j in range(i + 1, n):
                ans += not (i * j) % k and nums[i] == nums[j] 
        return ans

File: 2176. Count Equal and Divisible Pairs in an Array.cpp
class Solution {
public:
    int countPairs(vector<int>& nums, int k) 
    {
        int ans = 0, n = nums.size();
        for(int i = 0; i < n - 1; i++)
        {
            for(int j = i + 1; j < n; j++)
            {
                ans += (nums[i] == nums[j] and (i * j) % k == 0)
            }
        }
        return ans;
    }
};